reana_db/alembic/versions/20231002_1208_b85c3e601de4_separate_run_and_restart_number.py

```diff
@@ -0,0 +1,114 @@
+"""Separate run number into generation and restart number.
+
+Revision ID: b85c3e601de4
+Revises: 377cfbfccf75
+Create Date: 2023-10-02 12:08:18.292490
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "b85c3e601de4"
+down_revision = "377cfbfccf75"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Upgrade to b85c3e601de4 revision."""
+    # Add new columns (generation_number, restart_number)
+    op.add_column(
+        "workflow", sa.Column("generation_number", sa.Integer()), schema="__reana"
+    )
+    op.add_column(
+        "workflow",
+        sa.Column("restart_number", sa.Integer(), default=0),
```

Comment:
Regrading the new column names, I would call them `run_number_foo`, `run_number_bar` so that they would better express to semantic closeness to the concept of run_number.

As for "foo" and "bar", it could be perhaps:

- run_number_start, run_number_restart
- run_number_create, run_number_restart
- run_number_major, run_number_minor
- ...